sql/core/src/main/scala/org/apache/spark/sql/execution/local/LocalNode.scala

```diff
@@ -70,9 +70,14 @@ abstract class LocalNode(conf: SQLConf) extends TreeNode[LocalNode] with Logging
   def close(): Unit
 
   /**
+   * Returns the content through the [[Iterator]] interface.
+   */
+  final def asIterator: Iterator[InternalRow] = new LocalNodeIterator(this)
```

Comment:
Is this on the critical path? If yes - it'd actually be faster to expose the LocalNodeIterator type since JIT can inline it.